### PR TITLE
Add defaults for image-build.conf

### PR DESCRIPTION
**Pull request by lcarva**

### CODE REVIEW

File: atomic_reactor/plugins/pre_add_filesystem.py [hunk 1]
```diff
@@ -10,9 +10,17 @@
 try:
     # py2
     from ConfigParser import ConfigParser
+    # Although io.StringIO is available in python2, it
+    # always expects unicode during write. Since
+    # ConfigParser.ConfigParser does not support
+    # writing unicode, io cannot be used.
+    from StringIO import StringIO
```
Comment (vrutkovs) [inline atomic_reactor/plugins/pre_add_filesystem.py]: Importing ```BytesIO``` as ```StringIO``` might also work (I did that hack at https://github.com/projectatomic/atomic-reactor/blob/master/atomic_reactor/plugins/pre_add_yum_repo_by_url.py#L37)

File: atomic_reactor/plugins/pre_add_filesystem.py [hunk 2]
```diff
@@ -94,8 +124,13 @@ def parse_image_build_config(self, config_file_name):
         opts = {}
 
         config = ConfigParser()
+        config.readfp(self.get_default_image_build_conf())
         config.read(config_file_name)
 
+        config_str = StringIO()
+        config.write(config_str)
```
Comment (vrutkovs) [inline atomic_reactor/plugins/pre_add_filesystem.py]: We should use context manager here

File: atomic_reactor/plugins/pre_add_filesystem.py (hunk 1 above)
Comment (lcarva) [inline atomic_reactor/plugins/pre_add_filesystem.py]: I tried that, but it seems to not work if `from __future__ import unicode literals` is used.

File: atomic_reactor/plugins/pre_add_filesystem.py (hunk 2 above)
Comment (lcarva) [inline atomic_reactor/plugins/pre_add_filesystem.py]: `StringIO.StringIO` does not support context manager, only `io.StringIO`:

```
In [1]: import io

In [2]: with io.StringIO() as foo:
   ...:     pass
   ...: 

In [3]: import StringIO

In [4]: with StringIO.StringIO() as foo:
   ...:     pass
   ...: 
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-4c8d5c900858> in <module>()
----> 1 with StringIO.StringIO() as foo:
      2     pass
      3 

AttributeError: StringIO instance has no attribute '__exit__'

```

File: atomic_reactor/plugins/pre_add_filesystem.py (hunk 1 above)
Comment (vrutkovs) [inline atomic_reactor/plugins/pre_add_filesystem.py]: I think the trick was to read via ```BytesIO``` and write using ```StringIO``` due to a bug in py3, see ```pre_add_yum_repo_by_url.py``` code.

OTOH this is counter-intuitive, so probably we should keep StringIO

File: atomic_reactor/plugins/pre_add_filesystem.py (hunk 2 above)
Comment (vrutkovs) [inline atomic_reactor/plugins/pre_add_filesystem.py]: Right, lets leave it as it is then

**Comment by twaugh** [comment]
LGTM

**Pull request merged**